compiler/rustc_passes/src/loops.rs

```diff
@@ -399,4 +427,36 @@ impl<'hir> CheckLoopVisitor<'hir> {
             });
         }
     }
+
+    /// Is this a loop annotated with #[loop_match] that looks syntactically sound?
+    fn is_loop_match(
+        &self,
+        e: &'hir hir::Expr<'hir>,
+        body: &'hir hir::Block<'hir>,
+    ) -> Option<Label> {
+        if !self.tcx.hir_attrs(e.hir_id).iter().any(|attr| attr.has_name(sym::loop_match)) {
+            return None;
+        }
+
+        // NOTE: diagnostics are emitted during MIR construction.
```

Comment:
we need to do all of this matching again in mir construction, so I'm not emitting any diagnostics here